### DISCUSSION TITLE
display loading animation on request

### DISCUSSION THREAD

**Discussion by AlonsoK28** [digitalfortress-tech/typeahead-standalone#45] [Ideas]
indicate network activity using a loading a animation to improve user UX

this options is already available into jQuery Typeahead http://www.runningcoder.org/jquerytypeahead/documentation/

**Comment by AlonsoK28** [comment]
Is there currently any workaround to get the status of the request?

something equivalent to the following callbacks of [jQuery Typeahead](http://www.runningcoder.org/jquerytypeahead/documentation/) ?

- **onSendRequest (node, query)** | Gets called when the Ajax request(s) are sent. Either on initial requests or on every dynamic requests.
- **onReceiveRequest (node, query)** | Gets called when the Ajax request(s) are all received

This could be used to indicates network activity into your app an improve UX

**Reply by niketpathak** [reply]
Hello @AlonsoK28, starting with [v4.15.0](https://github.com/digitalfortress-tech/typeahead-standalone/releases/tag/v4.15.0), support for a `loader` template has been added. You are free to use text/img or any html as per your needs. 

Here's a small snippet that could be useful:

```js
/* typeahead config */
typeahead({
  // ... input, source, etc...
  templates: {
    loader: () => 'Loading...', /* Rendered while awaiting data from a remote source */
  }
});
```

See a live example on the [playground](https://typeahead.digitalfortress.tech/#playground?id=github-users-example) showing the loader in action. Hope that helps! 

Thanks!
